geopyspark/protobufregistry.py

```diff
@@ -0,0 +1,455 @@
+"""Contains the various encoding/decoding methods to bring values to/from Python from Scala."""
+from functools import partial
+import numpy as np
+from geopyspark.geopyspark_utils import check_environment
+check_environment()
+
+from geopyspark.geotrellis import (Extent, ProjectedExtent, TemporalProjectedExtent, SpatialKey,
+                                   SpaceTimeKey)
+
+from geopyspark.protobuf.tileMessages_pb2 import ProtoTile, ProtoMultibandTile, ProtoCellType
+from geopyspark.protobuf.extentMessages_pb2 import (ProtoExtent, ProtoProjectedExtent,
+                                                    ProtoTemporalProjectedExtent)
+from geopyspark.protobuf import keyMessages_pb2
+from geopyspark.protobuf import tupleMessages_pb2
+
+
+class ProtoBufRegistry(object):
```

Comment:
This can now become a module named `protobuf_codecs` with `_*_decoder` functions becoming `from_pb_tile` and reverse of `to_pb_tile`.